pkg/mesh/td.go

```diff
@@ -0,0 +1,227 @@
+func (td *TdSidecarEnv) loadFromEnv() {
+	env, present := os.LookupEnv("VPC_NETWORK_NAME")
+	if present {
+		td.NetworkName = env
+	}
+
+	env, present = os.LookupEnv("GCP_PROJECT_NUMBER")
+	if present {
+		td.ProjectNumber = env
+	}
+
+	env, present = os.LookupEnv("SERVICE_CIDR")
+	if present {
+		td.ServiceCidr = env
+	}
+
+	env, present = os.LookupEnv("ENVOY_PORT")
+	if present {
+		td.EnvoyPort = env
+	}
+
+	env, present = os.LookupEnv("ENVOY_ADMIN_PORT")
+	if present {
+		td.EnvoyAdminPort = env
+	}
+}
+
+// Currently zone/region is loaded for the cloud run instance using the metadata server.
+func (td *TdSidecarEnv) fetchZone() (string, error) {
+	zone, err := metadata.Get("instance/zone")
+	if err != nil {
+		return "", fmt.Errorf("could not determine zone from metadata server: ", err)
+	}
+	splits := strings.SplitAfter(zone, "/zones/")
+	if len(splits) != 2 {
+		return "", fmt.Errorf("could not determine zone from metadata server: ", err)
+	}
+	return splits[1], nil
+}
+
+func (td *TdSidecarEnv) fetchProjectNumber() (string, error) {
+	projectNumber, err := metadata.Get("project/numeric-project-id")
+	if err != nil {
+		return "", fmt.Errorf("could not determine project number from metadata server: ", err)
+	}
+	return projectNumber, nil
+}
+
+func (td *TdSidecarEnv) fetchNodeID() (string, error) {
+	hostName, hostErr := os.Hostname()
+	if hostErr != nil {
+		return "", hostErr
+	}
+	ips, ipErr := net.LookupIP(hostName)
+	if ipErr != nil {
+		return "", ipErr
+	}
+	return fmt.Sprintf("%s~%s", uuid.New().String(), ips[0].String()), nil
+}
+
+// Prepare TdSidecarEnv by using Metadata server and environment variables.
+func (td *TdSidecarEnv) prepare() {
+	// We use the project number that this cloud run service is running under as a default.
+	if projectNumber, err := td.fetchProjectNumber(); err != nil {
+		log.Println("Unable to auto-generate project_number.")
+	} else {
+		td.ProjectNumber = projectNumber
+	}
+
+	td.loadFromEnv()
+
+	if nodeID, err := td.fetchNodeID(); err != nil {
+		td.NodeID = fmt.Sprintf("%s~%s", uuid.New().String(), "127.0.0.1")
+		log.Println("Unable to generate proper nodeID, using: ", td.NodeID)
+	} else {
+		td.NodeID = nodeID
+	}
+
+	if zone, err := td.fetchZone(); err != nil {
+		td.EnvoyZone = "cloud-run-cluster"
+		log.Println("Unable to generate proper zone info, using: ", td.EnvoyZone)
+	} else {
+		td.EnvoyZone = zone
+	}
+}
+
+func (td *TdSidecarEnv) getIPTablesInterceptionEnvVars() []string {
+	envs := []string{
+		"TRAFFIC_DIRECTOR_GCE_VM_DEPLOYMENT_OVERRIDE=true",
+		"DISABLE_REDIRECTION_ON_LOCAL_LOOPBACK=true",
+		fmt.Sprintf("%s=%s", "ENVOY_DNS_PORT", td.EnvoyDnsPort),
+	}
+	return envs
+}
+
+func (td *TdSidecarEnv) logLevel() string {
+	return td.LogLevel
+}
+
+func (td *TdSidecarEnv) envoyPort() string {
+	return td.EnvoyPort
+}
+
+func (td *TdSidecarEnv) envoyAdminPort() string {
+	return td.EnvoyAdminPort
+}
+
+func (td *TdSidecarEnv) projectNumber() string {
```

Comment:
Not sure we need all those getters - any reason to not just use the field name directly ?

Reply:
No reason, I'm open to both. I added 376f53a21a3fc4ee6b187f051751af9850f4e409 to remove the getters.